clang/include/clang/Basic/LangOptions.h

```diff
@@ -816,6 +816,11 @@ class FPOptions {
     setAllowFPReassociate(LO.AllowFPReassoc);
     setNoHonorNaNs(LO.NoHonorNaNs);
     setNoHonorInfs(LO.NoHonorInfs);
+    // Ensure that if FiniteMathOnly is enabled, NoHonorNaNs and NoHonorInfs are
+    // also enabled. This is because FiniteMathOnly mode assumes no NaNs or Infs
+    // are present in computations.
+    assert((LO.FiniteMathOnly == (LO.NoHonorInfs && LO.NoHonorNaNs)) &&
```

Comment:
The `FiniteMathOnly` variable should be removed from `LangOptions` entirely at this point; the plan is to only rely on `NoHonorInfs` and `NoHonorNans` instead.

Reply:
There is no need for the assertion then. I suppose `CLFiniteMathOnly` should be removed too, right?

Reply:
Correct, that assertion can go away (don't forget to update the PR title and summary accordingly as well).

>  I suppose CLFiniteMathOnly should be removed too, right?

CC @AnastasiaStulova as OpenCL code owner.

I think `CLFiniteMathOnly` is a combination of `NoHonorInfs`, `NoHonorNaNs`, and `OpenCL` language mode. However, I don't see `CLFiniteMathOnly` being *used* by anything when I grep over the source. We set the flag from Options.td via marshaling, but we don't seem to ever check the value of the LangOpt anywhere. So it seems like `CLFiniteMathOnly` can be removed and maybe `-cl-finite-math-only` could be deprecated and eventually removed?

Reply:
Btw this option is defined in OpenCL spec so it would be better to keep this at least as a driver option.
https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#optimization-options

I think mapping it to ``NoHonorInfs``, ``NoHonorNaNs`` frontend options seems reasonable.

Can the description of this change be elaborated to describe what the intension is.

Tagging @svenvh for any additional feedback.

 But mapping

Reply:
@AnastasiaStulova thanks for the review.
The driver option `-cl-finite-math-only` can still be used.
Updated the PR's description.